manifests/server/passwd.pp

```diff
@@ -26,10 +26,11 @@
     #  without specifying a password ('ident' or 'trust' security). This is
     #  the default for pg_hba.conf.
     $escaped = postgresql::postgresql_escape($postgres_password)
+    $exec_command = "${shell_escape($psql_path)}${_dboption} -c \"ALTER ROLE \\\"${shell_escape($user)}\\\" PASSWORD \${NEWPASSWD_ESCAPED}\""
```

Comment:
So a few things happen here. First of all, `$_dboption` contains `$database` which is also user input. That should be escaped at line 19.

It's also important to realize this must use a shell because it passes the new password as an environment variable to hide the sensitive value. You could change `$_db_option` to an array at line 18 and then:
```suggestion
    $exec_command = [$psql_path] + $_dboption + ['-c', "ALTER ROLE \"${user}\" PASSWORD ${escaped}"]
```

But I don't know if if that properly generates a sensitive string. You'd also lose some clarity in what happens. So for now I'd escape the input in `$_dboption`.

Reply:
`$_dboption` has been escaped.
Have tried changing the full command to an array and it doesn't seem to parse cleanly